index.js

```diff
@@ -413,6 +413,105 @@ NYC.prototype.report = function () {
   }
 }
 
+// XXX(@isaacs) Index generation should move to istanbul-lib-processinfo
+NYC.prototype.writeProcessIndex = function () {
+  const dir = this.processInfoDirectory()
+  const pidToUid = new Map()
+  const infoByUid = new Map()
+  const eidToUid = new Map()
+  const infos = fs.readdirSync(dir).filter(f => f !== 'index.json').map(f => {
+    try {
+      const info = JSON.parse(fs.readFileSync(path.resolve(dir, f), 'utf-8'))
+      // on thiis first read pass, also map the pids to uuids
+      info.uuid = path.basename(f, '.json')
+      pidToUid.set(info.uuid, info.pid)
+      pidToUid.set(info.pid, info.uuid)
+      infoByUid.set(info.uuid, info)
+      if (info.externalId) {
+        eidToUid.set(info.externalId, info.uuid)
+      }
+      return info
+    } catch (er) {
+      return null
+    }
+  }).filter(Boolean)
+
+  // create all the parent-child links and write back the updated info
+  const needsUpdate = new Set()
+  infos.forEach(info => {
+    if (info.ppid && info.ppid !== '0' && !info.parent) {
+      info.parent = pidToUid.get(info.ppid)
+      needsUpdate.add(info)
+    }
+    if (info.parent) {
+      const parentInfo = infoByUid.get(info.parent)
+      if (parentInfo.children.indexOf(info.uuid) === -1) {
```

Comment:
Do we actually want to have `info.children` or `info.parent` in the per-process files?  This info will be written to `index.json`, I think we could completely avoid rewriting the per-process files.

Reply:
The current process tree generation is a bit brittle because it relies exclusively on pids and ppids, which are re-used.  Storing the uuid _somewhere_ felt important, but yeah, it is in the index.json already, so probably not necessary per-process, especially if it's the only reason to rewrite the files.

Reply:
This is a actually good point, especially if you're dealing with long running `watcher` process.  What if the `lib/process.js` constructor initialized `info.uuid`, then have `NYC.writeCoverageFile` just use that value instead of generating a new uuid?  Maybe even support something like `NYC_PROCESSINFO_PARENT_UUID` for `lib/process.js` to initialize `info.parent`?

Somewhat thinking out loud here, not sure how reasonable this is.

Reply:
Aha!  I was wrong, the current process prefixes the pids with the root uuid, so it's safe.  But still, using the parent uuid directly would be much clearer, imo.